src/sentry/web/frontend/auth_login.py

```diff
@@ -95,39 +243,238 @@ def get_register_form(self, request: Request, initial=None):
         next_uri_fallback = None
         if request.session.get("_next") is not None:
             next_uri_fallback = request.session.pop("_next")
         return request.GET.get(REDIRECT_FIELD_NAME, next_uri_fallback)
 
-    def get_post_register_url(self, request: Request):
-        base_url = auth.get_login_redirect(request)
-        params = {"frontend_events": json.dumps({"event_name": "Sign Up"})}
-        return add_params_to_url(base_url, params)
+    def handle_login_form_submit(self, request: Request, context: dict, **kwargs) -> HttpResponse:
+        """Validates a completed login  form, redirecting to the next
+        step or returning the form with its errors displayed."""
+        op = request.POST.get("op")
+        organization = kwargs.pop("organization", None)
+        login_form = AuthenticationForm(request, request.POST if op == "login" else None)
+
+        if self.is_ratelimited_login_attempt(request=request, login_form=login_form, op=op):
+            return self.get_ratelimited_login_form(
+                request=request, login_form=login_form, op=op, **kwargs
+            )
+        elif login_form.is_valid():
+            return self.redirect_to_next_login_step(
+                request=request, login_form=login_form, organization=organization
+            )
+        context.update({"login_form": login_form, "op": op or login})
+        return self.respond_login(request=request, context=context, **kwargs)
+
+    def is_ratelimited_login_attempt(
+        self, request: Request, login_form: AuthenticationForm, op: str
+    ) -> bool:
+        """Returns true if a user is attempting to login but is currently ratelimited."""
+        from sentry import ratelimits as ratelimiter
+        from sentry.utils.hashlib import md5_text
+
+        attempted_login = (
+            op == "login" and request.POST.get("username") and request.POST.get("password")
+        )
+
+        return attempted_login and ratelimiter.is_limited(
+            "auth:login:username:{}".format(
+                md5_text(login_form.clean_username(value=request.POST["username"])).hexdigest()
+            ),
+            limit=5,
+            window=60,  # 5 per minute should be enough for anyone
+        )
+
+    def get_ratelimited_login_form(
+        self, request: Request, login_form: AuthenticationForm, op: str, **kwargs
+    ) -> HttpResponse:
+        """Returns a login form with ratelimited errors displayed."""
+        login_form.errors["__all__"] = [
+            "You have made too many login attempts. Please try again later."
+        ]
+        metrics.incr("login.attempt", instance="rate_limited", skip_internal=True, sample_rate=1.0)
+
+        context = {
+            "op": op or "login",
+            "login_form": login_form,
+        }
+
+        context.update(additional_context.run_callbacks(request))
+        return self.respond_login(request=request, context=context, **kwargs)
+
+    def redirect_to_next_login_step(
```

Comment:
`_handle_login` is a method from the old `handle_basic_auth` and it may have strange dependencies - i flattened out the logic here, but would prefer to get rid of `_handle_login` when we get rid out `handle_basic_auth` in the next PR.